system/jhtree/jhinplace.cpp

```diff
@@ -1740,15 +1750,41 @@ void CJHInplaceTreeNode::load(CKeyHdr *_keyHdr, const void *rawData, offset_t _f
     if (numKeys)
     {
         size32_t len = hdr.keyBytes;
+        size32_t copyLen = len;
+        const byte * originalData = ((const byte *)rawData) + sizeof(hdr);
+        const byte * originalPayload = nullptr;
+        bool keepCompressedPayload = true;
+        if (isLeaf())
+        {
+            //Always calculate payload location so we can perform a consistency check later on.
+            originalPayload = queryPayload(originalData);
+            bool expandOnDemand = false;
```

Comment:
Is this for some possible future enhancement ?
Why declare expandOnDemand as false and then immediately and always check if its false ?

Reply:
Yes it is a reminder for a later change (HPCC-29559) - where I save the compressed data in memory and only expand when the payload is needed.